src/protocols/interactive_signing.rs

```diff
@@ -47,6 +39,35 @@ use crate::{
     },
 };
 
+/// Analogous to `Iterator::sum()`, but requires a non-empty iterator
+/// (so that it can be used for types with no `default()`, like `Ciphertext`)
```

Comment:
Not sure I understand this comment about `Default`. Wouldn't an impl Sum for CipherText work?

```
impl<P: PaillierParams> Sum for Ciphertext<P> {
    fn sum<I: Iterator<Item = Self>>(mut iter: I) -> Self {
        let mut result = iter.next().unwrap();
        for item in iter {
            result = result + item;
        }
        result
    }
}
```